cmd/dashboard.go

```diff
@@ -43,11 +46,39 @@ var DashboardCmd = &cobra.Command{
 			localPort = port
 		}
 
-		config, err := kubernetes.GetKubeConfig()
-
+		config, client, err := kubernetes.GetKubeConfigClient()
 		if err != nil {
 			print.FailureStatusEvent(os.Stdout, "Failed to initialize kubernetes client")
-			return
+			os.Exit(1)
+		}
+
+		// search for dashboard service namespace in order:
+		// user-supplied namespace, dapr-system, default
+		namespaces := []string{dashboardNamespace}
+		if dashboardNamespace != defaultNamespace {
+			namespaces = append(namespaces, defaultNamespace)
+		}
+		namespaces = append(namespaces, "default")
+
+		foundNamespace := ""
+		for _, namespace := range namespaces {
+			ok, _ := kubernetes.CheckPodExists(client, namespace, nil, dashboardSvc)
+			if ok {
+				foundNamespace = namespace
+				break
+			}
+		}
+
+		// if the service is not found, try to search all pods
```

Comment:
after some thoughts, I think we can remove this code path . This will give user wrong expectation that dashboard is running in namespace which user provided which could be false information . Exception is only for default and dapr-system.

Instead,  the correct experience would be , print the message that "Dapr dashboard found to be running in the namespace. Please try with namespace"